tests/integration/kubernetes/confidential_kbs.sh

```diff
@@ -419,13 +419,20 @@ function kbs_k8s_deploy() {
 	fi
 }
 
-# Return the kbs service host name in case ingress is configured
+# Return the kbs service public IP in case ingress is configured
 # otherwise the cluster IP.
 #
 kbs_k8s_svc_host() {
 	if kubectl get ingress -n "$KBS_NS" 2>/dev/null | grep -q kbs; then
-		kubectl get ingress "$KBS_INGRESS_NAME" -n "$KBS_NS" \
-			-o jsonpath='{.spec.rules[0].host}' 2>/dev/null
+		local host
+		# The ingress IP address can take a while to show up.
+		SECONDS=0
+		while true; do
+			host=$(kubectl get ingress "${KBS_INGRESS_NAME}" -n "${KBS_NS}" -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+			[[ -z "${host}" && ${SECONDS} -lt 30 ]] || break
+			sleep 5
```

Comment:
Do you need to do SECONDS += 5 here? At the moment I think it will loop forever if there is an issue with ingress coming up?

Reply:
Rá! This is a magic that I learned with @ldoktor while ago: that's a bash variable which is updated each time it's referenced. Quoting bash's manpage:

```
SECONDS
              Each time this parameter is referenced, it expands to the number
              of  seconds  since  shell invocation.  If a value is assigned to
              SECONDS, the value returned upon subsequent  references  is  the
              number  of seconds since the assignment plus the value assigned.
              The number of seconds at shell invocation and the  current  time
              are  always determined by querying the system clock.  If SECONDS
              is unset, it loses its special properties, even if it is  subse‐
              quently reset.
```